Add docs for dynamic privileges (#5077)
pingcap/docs: privilege-management.md

# Privilege Management

TiDB supports MySQL 5.7's privilege management system, including the syntax and privilege types. The following features from MySQL 8.0 are also supported:

* SQL Roles, starting with TiDB 3.0.

> * Dynamic privileges (experimental) <!-- include the version number after TiDB 5.0 -->

Reviewer: I don't think we need the 'experimental', we can try to make it GA recently.

Reviewer: Please remove the 'experimental' since we have #22778 merged.

+------------------------------------------------------------------+
```

### Dynamic privileges

> Since v5.1 TiDB features support dynamic privileges, a feature borrowed from MySQL 8.0. Dynamic privileges are intended to replace the `SUPER` privilege by implementing more fine-grained access to certain operations. For example, with dynamic privileges it is possible to create a user account that can only perform `BACKUP` and `RESTORE` operations.

Reviewer 2: ```suggestion
Since v5.1, TiDB features support dynamic privileges, a feature borrowed from MySQL 8.0. Dynamic privileges are intended to replace the `SUPER` privilege by implementing more fine-grained access to certain operations. For example, using dynamic privileges, system administrators can create a user account that can only perform `BACKUP` and `RESTORE` operations.
```

> <!-- include a "since" for the version number after TiDB 5.0 --> TiDB features support dynamic privileges, a feature borrowed from MySQL 8.0. Dynamic privileges are intended to replace the `SUPER` privilege by implementing more fine-grained access to certain operations. For example, with dynamic privileges it is possible to grant access to perform `BACKUP` and `RESTORE` operations but not `SHOW PROCESSLIST`.

Reviewer 2: + Can we say "Since v5.1" in the upcoming release-5.1 branch but remove it in master (because master is the latest by default)?
+ Could you specify "WHO" can grant access to perform `BACKUP` and `RESTORE`? With this information, the translated Chinese text will look more natural.
```suggestion
TiDB supports dynamic privileges, a feature borrowed from MySQL 8.0. Dynamic privileges are intended to replace the `SUPER` privilege by implementing more fine-grained access to certain operations. For example, with dynamic privileges enabled, it is possible to access to perform `BACKUP` and `RESTORE` operations but not `SHOW PROCESSLIST`.
```

Author: We can say since 5.1 - I was waiting for the versoin to be named.  I'll change it.

It should read:
"For example, with dynamic privileges ~enabled~, it is possible to grant.. "

Dynamic privileges are always enabled. The difference is if you chose to allocate a fine-grained privileges or not. Both are possible.

> Could you specify "WHO" can grant access to perform BACKUP and RESTORE? With this information, the translated Chinese text will look more natural.

It is possible for a system administrator to create a limited privilege account for use only in backup and restore operations. The backup and restore user might be part of a scripted process  - by limiting its access you help reduce the impact of privilege escalation scenarios (it is true for both scripted processes and users - you should always limit access to principal of least privilege.)

Dynamic privileges include:

* `BACKUP_ADMIN`
* `ROLE_ADMIN`
* `CONNECTION_ADMIN`
* `SYSTEM_VARIABLES_ADMIN`

> The full set of dynamic privileges can be observed with `SHOW PRIVILEGES`. Because plugins are permitted to add new privileges, it is possible that the list of privileges that are assignable will differ based on your TiDB installation.

Reviewer 2: ```suggestion
To see the full set of dynamic privileges, execute the `SHOW PRIVILEGES` statement. Because plugins are permitted to add new privileges, the list of privileges that are assignable might differ based on your TiDB installation.
```
